scripts/query_all_registrations.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+set -ueEo pipefail
+
+FETCH_ADDR_REGEX="^fetch[0-9a-z]{39}"
+RECONCILIATION_CONTRACT="${1-""}"
+
+if [[ ! $RECONCILIATION_CONTRACT =~ $FETCH_ADDR_REGEX ]]; then
+  echo -e "usage: query_all_registrations.sh CONTRACT
+    Where CONTRACT is the address of the reconciliation contract to query."
+  exit 1
+fi
+
+# Ensure reconciliation contract is paused
+RECONCILIATION_IS_PAUSED=$(fetchd query wasm contract-state smart "$RECONCILIATION_CONTRACT" '{"query_pause_status":{}}' --output json | jq -r '.data.paused')
+if [[ "$RECONCILIATION_IS_PAUSED" == "false" ]]; then
+  echo "Aborting!: reconciliation contract \"$RECONCILIATION_CONTRACT\" is not paused"
+  exit 1
+fi
+
+# Query reconciliation registrations
+fetchd query wasm contract-state smart "$RECONCILIATION_CONTRACT" '{"query_all_registrations":{}}' --output json | jq '.data.registrations'
```

Comment:
not sure we want this in the repo, as the contract is getting deleted by migration, so this won't work anymore

Reply:
ah good point! I was seeing this from a migration and post-migration perspective only, where we'll provide a reconciliation state dump to be used at migration time, and where we can't really dump it at migration time because the network will be stopped, preventing any query to land. 

But yes, it's best to allow everyone to recreate all our dumps and data if they want to, and this could be done as a "pre-migration" step in between the contract pause and the network shutdown and this script can help here. So I'd vote to keep it, but in the migration doc repo (just for the shake of having all external scripts in a single repo with paths relative to the procedure readme so people don't get lost cd'ing around) - and the extra pause check will be handy there!

Reply:
I pushed this script to the [reconciliation/query_all_registrations](https://github.com/fetchai/fetchd/commits/reconciliation/query_all_registrations) branch so that it doesn't get lost.